app/ui/browser/views/navbar/location.jsx

```diff
@@ -306,7 +322,14 @@ export class Location extends Component {
           ref="input"
           onFocus={this.handleURLBarFocus}
           onBlur={this.handleURLBarBlur}
-          onChange={this.props.onLocationChange}
+          onChange={(e) => {
+            // There's a weird case here where the location can be reset from an action
```

Comment:
we discussed this in person, the comment should be expanded to explain why there is a weird case